app/models/vote.rb

```diff
@@ -29,8 +29,12 @@ def self.noes
     where("votes.vote = 'no'")
   end
 
-  def rebellion?
-    !free? && vote != whip_guess
+  def rebellion?(whip = nil)
+    if whip.nil?
+      !free? && vote != whip_guess
+    else
+      !whip.free? && vote != whip.whip_guess
+    end
```

Comment:
@mario-areias wrote on outdated diff:

>Sure. But what do you think about this method overall? I did that to improve the performance, but I hated the end resutlts. I don't have complete understanding about what this method does, so I'm not sure how to improve it. If you have any suggestions, that would be awesome!

How does it improve performance?

I'm not really sure this is doing what you think it's doing. In the first version of the `if` statement it's using the [delegated methods](https://github.com/mario-areias/publicwhip/blob/fc010657f191f1d50601e18a4b84096f8ca0431f/app/models/vote.rb#L6) and the second version it's just chaining the methods onto the [`#whip` method](https://github.com/mario-areias/publicwhip/blob/fc010657f191f1d50601e18a4b84096f8ca0431f/app/models/vote.rb#L9-L11) directly. I can't see how there would be a difference between the two.